client/driver/docker.go

```diff
@@ -333,6 +333,18 @@ func (d *DockerDriver) Open(ctx *ExecContext, handleID string) (DriverHandle, er
 	return h, nil
 }
 
+// dockerClient returns a configured *docker.Client from the ClientConfig
+func (d *DockerDriver) dockerClient() (*docker.Client, error) {
+	dockerEndpoint := d.config.Read("docker.endpoint")
+	if dockerEndpoint == "" {
+		client, err := docker.NewClientFromEnv()
+		return client, err
+	} else {
+		client, err := docker.NewClient(dockerEndpoint)
+		return client, err
```

Comment:
You can collapse these even further by just doing a `return docker.NewClient(...)` etc.